modules/host/contractmanager/storagefoldershrink_bench_test.go

```diff
@@ -0,0 +1,38 @@
+package contractmanager
+
+import (
+	"testing"
+)
+
+// BenchmarkUnsetManySectors checks that unsetting a bunch of sectors
+// individually takes an acceptable amount of time. Millions should be possible
+// in well under a second.
+//
+// My laptop is doing 10e6 in 1.2 seconds. This is on the edge of too slow, but
+// overall it's close enough.
+func BenchmarkUnsetManySectors(b *testing.B) {
+	// Create a uint64 that has all of the bits set.
+	base := uint64(0)
+	base--
+
+	// Benchmark how long it takes to unset the bits.
+	for i := 0; i < b.N; i++ {
+		// Create a usage array with all bits set.
+		b.StopTimer()
+		usageArray := make([]uint64, 10e6)
+		for i := 0; i < len(usageArray); i++ {
+			usageArray[i] = base
+		}
+		b.StartTimer()
```

Comment:
can probably stick this outside the loop (and add `b.ResetTimer()` after). The speed of the `&` operation shouldn't be affected by whether the bits are set or unset. So perhaps you don't need to set each element to `base` either.